Automattic/sensei
# Integrate tour-kit library, implement Course and Lesson tours

Resolves https://github.com/Automattic/sensei/issues/7444
And https://github.com/Automattic/sensei/issues/7477
Part of https://github.com/Automattic/sensei/issues/7494


## Proposed Changes
* Integrated [tour-kit](https://www.npmjs.com/package/@automattic/tour-kit) library to show sensei tours
* Added Tours for Courses and Lessons

### Separate Issues:
I'll create separate mini issues to deal with the following ones (and anything else we notice while testing this) to not deal with them here in this PR and enable working on other issues that are dependant on the tour kit. As this feature is behind a feature flag, we can go ahead and merge it if it looks good overall, and can work the other issues separately. The main goal of this PR is integrating the tour-kit library properly.

- We don't deal with the loading logic here. We are loading Course and Lesson tours for all Courses and Lessons
- When a tour is closed by clicking on the cross, we don't persist that data for that user, so it'll load again on next reload
- Show tour only after modal is closed (Update - later I implemented this and also introduced a redux store, to fix an e2e issue. it still needs some work and I'll also add tests under the dedicated issue for this https://github.com/Automattic/sensei/issues/7494)
- When content drip is enabled, tour is shown over the content drip area below.

## Testing Instructions


1. Make sure your local environment is set up as 'development' [env](https://developer.wordpress.org/reference/functions/wp_get_environment_type/). Otherwise, this feature won't be enabled
2. Run `composer install`
3. Run `npm install`
4. Make sure you're running "npm run start" or have run "npm run build"
5. Create a new Course and make sure you can take the tour after adding a Course Outline block, minimize it and cross it
6. Create a new lesson and make sure you see the tour after adding a Quiz block, minimize it and cross it

#### Course Tour
https://github.com/Automattic/sensei/assets/6820724/825eef9e-45b0-4f85-85bf-74f8c08e82fe

#### Lesson Tour
https://github.com/Automattic/sensei/assets/6820724/b152b239-dc8b-4962-a767-0ed4d1411f03



## New/Updated Hooks


* `sensei_tour_loaders` - Loads a sensei tour script

## Pre-Merge Checklist
